### PR TITLE
Jefferson_Assignment_10

### PR DESCRIPTION
**Pull request by jeffft9**
Hello Aishwarya and Aakash, please review the code as per professor's specs. Thanks

### CODE REVIEW

File: Wordle.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+            if(not Ui.checkWordLength(input_word)):
+                continue
+
+            # check if word was entered previously
+            if(input_word not in word_list):
+                print("\nWARNING : Entered word not in dictionary!")
+                continue
+
+            # check if the guess is correct
+            if input_word == todays_word:
```
Comment (TheAakash1886) [inline Wordle.py:135]: Can also be made into a function to be consistent with the design philosophy and be easier to understand as it separates each use case to make code easier to read

File: Test.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    assert(result == False)
+
+
+def test_word_length_three():
+    result = Ui.checkWordLength(
+        "SONAR")
+    assert(result == True)
+
+
+def test_word_length_three():
```
Comment (TheAakash1886) [inline Test.py:77]: Some test cases have the same name which might make it difficult to pinpoint which specific test case failed

**Review by TheAakash1886 — COMMENTED** [review]
Please read and review the comments and make changes when felt necessary.

File: Dictionary.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+class Dictionary:
+    def __init__(self) -> None:
+        self.myWordList = []
+        self.finalWord = ""
+
+    def __str__(self) -> str:
+        return str(self.finalWord)
+
+    # Getter
+    def getter(self):
```
Comment (aishwaryashirbhate) [inline Dictionary.py:14]: Please add appropriate type hints for this function.

File: Dictionary.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import random
+from typing import List, Tuple
+
+
+class Dictionary:
```
Comment (aishwaryashirbhate) [inline Dictionary.py:5]: Please add appropriate comments for this class mentioning its function or usage.

File: Helper.py [hunk 5]
```diff
@@ -0,0 +1,132 @@
+from LinkedList import Node
+from LinkedList import SLinkedList
+
+
+class Helper:
```
Comment (aishwaryashirbhate) [inline Helper.py:5]: Type hints missing. Please add type hints and comments for the function under this class.

File: Helper.py [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+            copyTentative = tentative[:]
+
+            if(len(positionWords) == 5):
+                for x in tentative:
+                    for index in range(5):
+                        if positionWords[index] != " " and positionWords[index] != x[index]:
+                            copyTentative.remove(x)
+                            break
+
+            # linkedList = SLinkedList()
```
Comment (aishwaryashirbhate) [inline Helper.py:92]: Please remove commented out redundant code.

File: Test.py [hunk 7]
```diff
@@ -0,0 +1,109 @@
+import pytest
+
+# from UI import *
+# from Dictionary import *
+from UI import Ui
+import Dictionary as dictionaryClass
+from Wordle import Wordle
+
+
+def test_both_empty():
```
Comment (aishwaryashirbhate) [inline Test.py:10]: Add comments for all test cases mentioning which function it is covering.

File: UI.py [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    def checkWordUsed(input_word: str, entered_words: List[str]) -> bool:
+        if input_word in entered_words:
+            print(f'\nWARNING : {input_word} already entered once!!!')
+            return False
+        else:
+            return True
+
+    # check if entered word is of length 5
+
+    def checkWordLength(input_word: str) -> bool:
```
Comment (aishwaryashirbhate) [inline UI.py:33]: Good job in mentioning comments, type hints and good usage of meaningful variable names.

File: Wordle.py [hunk 9]
```diff
@@ -0,0 +1,226 @@
...
+        except IOError:
+            print('An error occured trying to read the file.')
+            print(
+                'Please make sure "gameplay.log" is present in the directory before running the program')
+            quit()
+        except ZeroDivisionError:
+            f.write(f"Win % : 0\n\n")
+
+
+if __name__ == "__main__":
```
Comment (aishwaryashirbhate) [inline Wordle.py:222]: Add comments stating main function and its usage.

File: Wordle.py [hunk 10]
```diff
@@ -0,0 +1,226 @@
...
+                todays_word, word_list, entered_words, answer, attempt, words_used, good_letters, bad_letters = self.initVariables(
+                    words_used)
+                # print("Starting a New Game, Press enter to exit")
+                break
+
+    def displayResults(self, games_played: int, games_won: int, guess_history: List[str]) -> None:
+        print(f"\nNumber of Games Played : {games_played}")
+        print(f"Guess distribution : {guess_history}")
+
+        try:
```
Comment (aishwaryashirbhate) [inline Wordle.py:202]: Good job in writing appropriate functions try catch functions.

**Review by aishwaryashirbhate — CHANGES_REQUESTED** [review]
Hello Jefferson,
Please review the comments and perform the necessary code changes.

Best,
Aishwarya

**Pull request merged**